File: create_presentation.php
<?php

require 'vendor/autoload.php';

use PhpOffice\PhpPresentation\PhpPresentation;
use PhpOffice\PhpPresentation\IOFactory;
use PhpOffice\PhpPresentation\Style\Color;
use PhpOffice\PhpPresentation\Style\Alignment;
use PhpOffice\PhpPresentation\Style\Bullet;
use PhpOffice\PhpPresentation\Slide\Background\Color as BgColor;

// Create new PHPPresentation object
$objPHPPresentation = new PhpPresentation();

// Set properties
$objPHPPresentation->getDocumentProperties()
    ->setCreator("Office Archive Management")
    ->setLastModifiedBy("System")
    ->setTitle("Office Archive Management System")
    ->setSubject("Presentasi Aplikasi Manajemen Arsip")
    ->setDescription('Presentasi untuk Office Archive Management System')
    ->setKeywords('office archive management system')
    ->setCategory('Presentasi');

// Set default font
$objPHPPresentation->getDefaultStyle()
    ->getFont()->setName('Arial')
    ->setSize(10);

// Define colors
$primaryColor = new Color('2B579A');
$secondaryColor = new Color('4472C4');
$accentColor = new Color('ED7D31');

// ===== SLIDE 1: TITLE =====
$currentSlide = $objPHPPresentation->getActiveSlide();
$currentSlide->getBackground()->setFillType(new BgColor())->setColor($primaryColor);

// Add title
$shape = $currentSlide->createRichTextShape()
    ->setHeight(100)
    ->setWidth(900)
    ->setOffsetX(10)
    ->setOffsetY(150);
$shape->getActiveParagraph()->getAlignment()->setHorizontal(Alignment::HORIZONTAL_CENTER);
$textRun = $shape->createTextRun('Office Archive Management System');
$textRun->getFont()
    ->setBold(true)
    ->setSize(36)
    ->setColor(new Color('FFFFFF'));

// Add subtitle
$shape = $currentSlide->createRichTextShape()
    ->setHeight(60)
    ->setWidth(900)
    ->setOffsetX(10)
    ->setOffsetY(250);
$shape->getActiveParagraph()->getAlignment()->setHorizontal(Alignment::HORIZONTAL_CENTER);
$textRun = $shape->createTextRun('Solusi Digital untuk Manajemen Dokumen Perkantoran yang Efisien');
$textRun->getFont()
    ->setSize(20)
    ->setColor(new Color('FFFFFF'));

// Add footer
$shape = $currentSlide->createRichTextShape()
    ->setHeight(30)
    ->setWidth(600)
    ->setOffsetX(10)
    ->setOffsetY(500);
$textRun = $shape->createTextRun('Presentasi - ' . date('d F Y'));
$textRun->getFont()
    ->setSize(12)
    ->setColor(new Color('FFFFFF'));

// ===== SLIDE 2: AGENDA =====
$currentSlide = $objPHPPresentation->createSlide();
$this->addTitle($currentSlide, 'Agenda');

$agendaItems = [
    'Gambaran Umum Sistem',
    'Fitur Utama Aplikasi',
    'Teknologi yang Digunakan',
    'Demo Aplikasi',
    'Keunggulan Solusi',
    'Rencana Pengembangan'
];

$this->addBulletPoints($currentSlide, $agendaItems, 150);

// ===== SLIDE 3: GAMBARAN UMUM =====
$currentSlide = $objPHPPresentation->createSlide();
$this->addTitle($currentSlide, 'Gambaran Umum');

$points = [
    'Platform manajemen dokumen digital terintegrasi',
    'Menyediakan solusi pengarsipan yang terstruktur',
    'Mendukung berbagai format dokumen',
    'Memudahkan pelacakan dan pencarian dokumen',
    'Mengotomatiskan alur kerja persetujuan dokumen'
];

$this->addBulletPoints($currentSlide, $points, 150);

// ===== SLIDE 4: FITUR UTAMA 1 =====
$currentSlide = $objPHPPresentation->createSlide();
$this->addTitle($currentSlide, 'Fitur Utama - Manajemen Dokumen');

$points = [
    'Upload berbagai format file (PDF, DOC, XLS, dll)',
    'Pencarian canggih dengan filter',
    'Preview dokumen langsung di browser',
    'Riwayat perubahan dan versi dokumen',
    'Manajemen kategori terstruktur'
];

$this->addBulletPoints($currentSlide, $points, 150);

// ===== SLIDE 5: FITUR UTAMA 2 =====
$currentSlide = $objPHPPresentation->createSlide();
$this->addTitle($currentSlide, 'Fitur Utama - SPK & PPBJ');

$points = [
    'Form pengajuan online yang mudah digunakan',
    'Proses persetujuan multi-level',
    'Notifikasi real-time',
    'Pelacakan status pengajuan',
    'Arsip digital dokumen yang disetujui'
];

$this->addBulletPoints($currentSlide, $points, 150);

// ===== SLIDE 6: TEKNOLOGI =====
$currentSlide = $objPHPPresentation->createSlide();
$this->addTitle($currentSlide, 'Teknologi yang Digunakan');

$tech = [
    'Backend: PHP 8.1+, Laravel 10.x, MySQL 8.0+',
    'Frontend: Bootstrap 5, jQuery, DataTables',
    'Keamanan: Autentikasi multi-faktor, Enkripsi data, Audit log',
    'Hosting: Dapat di-host di berbagai platform cloud'
];

$this->addBulletPoints($currentSlide, $tech, 150);

// ===== SLIDE 7: DEMO =====
$currentSlide = $objPHPPresentation->createSlide();
$this->addTitle($currentSlide, 'Demo Aplikasi');

$points = [
    'Login dan Dashboard',
    'Manajemen Dokumen',
    'Pengajuan SPK & PPBJ',
    'Manajemen Pengguna',
    'Laporan dan Analitik'
];

$this->addBulletPoints($currentSlide, $points, 150, 32);

// ===== SLIDE 8: KEUNGGULAN =====
$currentSlide = $objPHPPresentation->createSlide();
$this->addTitle($currentSlide, 'Keunggulan Solusi');

$points = [
    'Efisiensi waktu dengan pengelolaan dokumen terpusat',
    'Akses mudah kapan saja, di mana saja',
    'Keamanan data terjamin dengan enkripsi',
    'Antarmuka yang ramah pengguna',
    'Dapat disesuaikan dengan kebutuhan bisnis'
];

$this->addBulletPoints($currentSlide, $points, 150);

// ===== SLIDE 9: RENCANA PENGEMBANGAN =====
$currentSlide = $objPHPPresentation->createSlide();
$this->addTitle($currentSlide, 'Rencana Pengembangan');

$points = [
    'Integrasi dengan layanan cloud storage',
    'Pengembangan aplikasi mobile',
    'Fitur tanda tangan digital',
    'Analitik dan pelaporan yang lebih canggih',
    'Integrasi dengan sistem eksternal'
];

$this->addBulletPoints($currentSlide, $points, 150);

// ===== SLIDE 10: PENUTUP =====
$currentSlide = $objPHPPresentation->createSlide();
$currentSlide->getBackground()->setFillType(new BgColor())->setColor($primaryColor);

// Add title
$shape = $currentSlide->createRichTextShape()
    ->setHeight(100)
    ->setWidth(900)
    ->setOffsetX(10)
    ->setOffsetY(150);
$shape->getActiveParagraph()->getAlignment()->setHorizontal(Alignment::HORIZONTAL_CENTER);
$textRun = $shape->createTextRun('Terima Kasih');
$textRun->getFont()
    ->setBold(true)
    ->setSize(48)
    ->setColor(new Color('FFFFFF'));

// Add subtitle
$shape = $currentSlide->createRichTextShape()
    ->setHeight(60)
    ->setWidth(900)
    ->setOffsetX(10)
    ->setOffsetY(250);
$shape->getActiveParagraph()->getAlignment()->setHorizontal(Alignment::HORIZONTAL_CENTER);
$textRun = $shape->createTextRun('Ada pertanyaan?');
$textRun->getFont()
    ->setSize(28)
    ->setColor(new Color('FFFFFF'));

// Add contact info
$contacts = [
    '📧 info@perusahaananda.com',
    '📱 +62 822-8721-8274',
    '🌐 www.perusahaananda.com'
];

$yPos = 350;
foreach ($contacts as $contact) {
    $shape = $currentSlide->createRichTextShape()
        ->setHeight(40)
        ->setWidth(600)
        ->setOffsetX(160)
        ->setOffsetY($yPos);
    $shape->getActiveParagraph()->getAlignment()->setHorizontal(Alignment::HORIZONTAL_LEFT);
    $textRun = $shape->createTextRun($contact);
    $textRun->getFont()
        ->setSize(20)
        ->setColor(new Color('FFFFFF'));
    
    $yPos += 50;
}

// Add footer
$shape = $currentSlide->createRichTextShape()
    ->setHeight(20)
    ->setWidth(900)
    ->setOffsetX(10)
    ->setOffsetY(550);
$shape->getActiveParagraph()->getAlignment()->setHorizontal(Alignment::HORIZONTAL_CENTER);
$textRun = $shape->createTextRun('© 2024 Office Archive Management System. All rights reserved.');
$textRun->getFont()
    ->setSize(10)
    ->setColor(new Color('CCCCCC'));

// Save file
$oWriterPPTX = IOFactory::createWriter($objPHPPresentation, 'PowerPoint2007');
$oWriterPPTX->save('presentasi_office_archive.pptx');

echo "Presentasi berhasil dibuat: presentasi_office_archive.pptx";

/**
 * Helper function to add title to slide
 */
function addTitle($slide, $title) {
    $shape = $slide->createRichTextShape()
        ->setHeight(50)
        ->setWidth(900)
        ->setOffsetX(10)
        ->setOffsetY(50);
    $shape->getActiveParagraph()->getAlignment()->setHorizontal(Alignment::HORIZONTAL_LEFT);
    $textRun = $shape->createTextRun($title);
    $textRun->getFont()
        ->setBold(true)
        ->setSize(32)
        ->setColor(new Color('2B579A'));
}

/**
 * Helper function to add bullet points to slide
 */
function addBulletPoints($slide, $items, $yStart = 100, $fontSize = 24) {
    $yPos = $yStart;
    
    foreach ($items as $item) {
        $shape = $slide->createRichTextShape()
            ->setHeight(40)
            ->setWidth(800)
            ->setOffsetX(100)
            ->setOffsetY($yPos);
        
        $shape->getActiveParagraph()
            ->getBulletStyle()
            ->setBulletType(Bullet::TYPE_BULLET)
            ->setBulletChar('•');
            
        $textRun = $shape->createTextRun($item);
        $textRun->getFont()
            ->setSize($fontSize)
            ->setColor(new Color('000000'));
        
        $yPos += 50;
    }
}

File: create_presentation.py
from pptx import Presentation
from pptx.util import Inches, Pt
from pptx.enum.text import PP_ALIGN, MSO_ANCHOR
from pptx.dml.color import RGBColor
from datetime import datetime

def create_presentation():
    # Create presentation object
    prs = Presentation()
    
    # Set slide dimensions (16:9 widescreen)
    prs.slide_width = Inches(13.333)
    prs.slide_height = Inches(7.5)
    
    # Define colors
    PRIMARY_COLOR = RGBColor(59, 89, 152)  # Dark blue
    SECONDARY_COLOR = RGBColor(66, 133, 244)  # Light blue
    ACCENT_COLOR = RGBColor(255, 153, 0)  # Orange
    WHITE = RGBColor(255, 255, 255)
    
    # ===== SLIDE 1: TITLE =====
    slide_layout = prs.slide_layouts[0]  # Title slide
    slide = prs.slides.add_slide(slide_layout)
    
    # Set background
    background = slide.background
    fill = background.fill
    fill.solid()
    fill.fore_color.rgb = PRIMARY_COLOR
    
    # Title
    title = slide.shapes.title
    title.text = "Office Archive Management System"
    title.text_frame.paragraphs[0].font.color.rgb = WHITE
    title.text_frame.paragraphs[0].font.size = Pt(44)
    title.text_frame.paragraphs[0].alignment = PP_ALIGN.CENTER
    
    # Subtitle
    subtitle = slide.placeholders[1]
    subtitle.text = "Solusi Digital untuk Manajemen Dokumen Perkantoran yang Efisien"
    subtitle.text_frame.paragraphs[0].font.color.rgb = WHITE
    subtitle.text_frame.paragraphs[0].font.size = Pt(20)
    
    # Footer
    footer = slide.shapes.add_textbox(Inches(0.5), Inches(6.5), Inches(12), Inches(1))
    tf = footer.text_frame
    p = tf.add_paragraph()
    p.text = "Presentasi - " + datetime.now().strftime("%d %B %Y")
    p.font.size = Pt(12)
    p.font.color.rgb = WHITE
    p.alignment = PP_ALIGN.LEFT
    
    # ===== SLIDE 2: AGENDA =====
    slide = prs.slides.add_slide(prs.slide_layouts[1])
    title = slide.shapes.title
    title.text = "Agenda"
    
    content = [
        "Gambaran Umum Sistem",
        "Fitur Utama Aplikasi",
        "Teknologi yang Digunakan",
        "Demo Aplikasi",
        "Keunggulan Solusi",
        "Rencana Pengembangan"
    ]
    
    body_shape = slide.shapes.placeholders[1]
    tf = body_shape.text_frame
    
    for item in content:
        p = tf.add_paragraph()
        p.text = f"• {item}"
        p.level = 0
        p.font.size = Pt(24)
        p.space_after = Pt(12)
    
    # ===== SLIDE 3: GAMBARAN UMUM =====
    slide = prs.slides.add_slide(prs.slide_layouts[1])
    title = slide.shapes.title
    title.text = "Gambaran Umum"
    
    points = [
        "Platform manajemen dokumen digital terintegrasi",
        "Menyediakan solusi pengarsipan yang terstruktur",
        "Mendukung berbagai format dokumen",
        "Memudahkan pelacakan dan pencarian dokumen",
        "Mengotomatiskan alur kerja persetujuan dokumen"
    ]
    
    body_shape = slide.shapes.placeholders[1]
    tf = body_shape.text_frame
    
    for point in points:
        p = tf.add_paragraph()
        p.text = point
        p.level = 0
        p.font.size = Pt(24)
        p.space_after = Pt(12)
    
    # ===== SLIDE 4: FITUR UTAMA 1 =====
    slide = prs.slides.add_slide(prs.slide_layouts[1])
    title = slide.shapes.title
    title.text = "Fitur Utama - Manajemen Dokumen"
    
    points = [
        "Upload berbagai format file (PDF, DOC, XLS, dll)",
        "Pencarian canggih dengan filter",
        "Preview dokumen langsung di browser",
        "Riwayat perubahan dan versi dokumen",
        "Manajemen kategori terstruktur"
    ]
    
    body_shape = slide.shapes.placeholders[1]
    tf = body_shape.text_frame
    
    for point in points:
        p = tf.add_paragraph()
        p.text = f"• {point}"
        p.level = 0
        p.font.size = Pt(24)
        p.space_after = Pt(12)
    
    # ===== SLIDE 5: FITUR UTAMA 2 =====
    slide = prs.slides.add_slide(prs.slide_layouts[1])
    title = slide.shapes.title
    title.text = "Fitur Utama - SPK & PPBJ"
    
    points = [
        "Form pengajuan online yang mudah digunakan",
        "Proses persetujuan multi-level",
        "Notifikasi real-time",
        "Pelacakan status pengajuan",
        "Arsip digital dokumen yang disetujui"
    ]
    
    body_shape = slide.shapes.placeholders[1]
    tf = body_shape.text_frame
    
    for point in points:
        p = tf.add_paragraph()
        p.text = f"• {point}"
        p.level = 0
        p.font.size = Pt(24)
        p.space_after = Pt(12)
    
    # ===== SLIDE 6: TEKNOLOGI =====
    slide = prs.slides.add_slide(prs.slide_layouts[1])
    title = slide.shapes.title
    title.text = "Teknologi yang Digunakan"
    
    tech = [
        ("Backend", "PHP 8.1+, Laravel 10.x, MySQL 8.0+"),
        ("Frontend", "Bootstrap 5, jQuery, DataTables"),
        ("Keamanan", "Autentikasi multi-faktor, Enkripsi data, Audit log"),
        ("Hosting", "Dapat di-host di berbagai platform cloud")
    ]
    
    body_shape = slide.shapes.placeholders[1]
    tf = body_shape.text_frame
    
    for category, items in tech:
        p = tf.add_paragraph()
        p.text = f"{category}:"
        p.level = 0
        p.font.bold = True
        p.font.size = Pt(22)
        p.space_after = Pt(4)
        
        p = tf.add_paragraph()
        p.text = items
        p.level = 1
        p.font.size = Pt(20)
        p.space_after = Pt(16)
    
    # ===== SLIDE 7: DEMO =====
    slide = prs.slides.add_slide(prs.slide_layouts[1])
    title = slide.shapes.title
    title.text = "Demo Aplikasi"
    
    points = [
        "Login dan Dashboard",
        "Manajemen Dokumen",
        "Pengajuan SPK & PPBJ",
        "Manajemen Pengguna",
        "Laporan dan Analitik"
    ]
    
    body_shape = slide.shapes.placeholders[1]
    tf = body_shape.text_frame
    
    for point in points:
        p = tf.add_paragraph()
        p.text = f"• {point}"
        p.level = 0
        p.font.size = Pt(28)
        p.space_after = Pt(12)
    
    # ===== SLIDE 8: KEUNGGULAN =====
    slide = prs.slides.add_slide(prs.slide_layouts[1])
    title = slide.shapes.title
    title.text = "Keunggulan Solusi"
    
    points = [
        "Efisiensi waktu dengan pengelolaan dokumen terpusat",
        "Akses mudah kapan saja, di mana saja",
        "Keamanan data terjamin dengan enkripsi",
        "Antarmuka yang ramah pengguna",
        "Dapat disesuaikan dengan kebutuhan bisnis"
    ]
    
    body_shape = slide.shapes.placeholders[1]
    tf = body_shape.text_frame
    
    for point in points:
        p = tf.add_paragraph()
        p.text = f"• {point}"
        p.level = 0
        p.font.size = Pt(24)
        p.space_after = Pt(12)
    
    # ===== SLIDE 9: RENCANA PENGEMBANGAN =====
    slide = prs.slides.add_slide(prs.slide_layouts[1])
    title = slide.shapes.title
    title.text = "Rencana Pengembangan"
    
    points = [
        "Integrasi dengan layanan cloud storage",
        "Pengembangan aplikasi mobile",
        "Fitur tanda tangan digital",
        "Analitik dan pelaporan yang lebih canggih",
        "Integrasi dengan sistem eksternal"
    ]
    
    body_shape = slide.shapes.placeholders[1]
    tf = body_shape.text_frame
    
    for point in points:
        p = tf.add_paragraph()
        p.text = f"• {point}"
        p.level = 0
        p.font.size = Pt(24)
        p.space_after = Pt(12)
    
    # ===== SLIDE 10: PENUTUP =====
    slide = prs.slides.add_slide(prs.slide_layouts[5])  # Blank layout
    
    # Add title
    title_box = slide.shapes.add_textbox(Inches(1), Inches(1), Inches(11.33), Inches(1.5))
    tf = title_box.text_frame
    p = tf.add_paragraph()
    p.text = "Terima Kasih"
    p.font.size = Pt(54)
    p.font.bold = True
    p.font.color.rgb = PRIMARY_COLOR
    p.alignment = PP_ALIGN.CENTER
    
    # Add subtitle
    subtitle_box = slide.shapes.add_textbox(Inches(1), Inches(2.5), Inches(11.33), Inches(1))
    tf = subtitle_box.text_frame
    p = tf.add_paragraph()
    p.text = "Ada pertanyaan?"
    p.font.size = Pt(32)
    p.font.color.rgb = SECONDARY_COLOR
    p.alignment = PP_ALIGN.CENTER
    
    # Add contact info
    contact_info = [
        ("📧", "Email", "info@perusahaananda.com"),
        ("📱", "Telepon/WA", "+62 822-8721-8274"),
        ("🌐", "Website", "www.perusahaananda.com")
    ]
    
    y_pos = 4.0
    for icon, label, value in contact_info:
        # Icon
        icon_box = slide.shapes.add_textbox(Inches(4), Inches(y_pos), Inches(0.5), Inches(0.5))
        tf = icon_box.text_frame
        p = tf.add_paragraph()
        p.text = icon
        p.font.size = Pt(20)
        
        # Label
        label_box = slide.shapes.add_textbox(Inches(4.7), Inches(y_pos), Inches(2), Inches(0.5))
        tf = label_box.text_frame
        p = tf.add_paragraph()
        p.text = f"{label}:"
        p.font.size = Pt(18)
        p.font.bold = True
        
        # Value
        value_box = slide.shapes.add_textbox(Inches(6), Inches(y_pos), Inches(4), Inches(0.5))
        tf = value_box.text_frame
        p = tf.add_paragraph()
        p.text = value
        p.font.size = Pt(18)
        
        y_pos += 0.6
    
    # Add footer
    footer = slide.shapes.add_textbox(Inches(0.5), Inches(6.8), Inches(12), Inches(0.5))
    tf = footer.text_frame
    p = tf.add_paragraph()
    p.text = "© 2024 Office Archive Management System. All rights reserved."
    p.font.size = Pt(10)
    p.font.color.rgb = RGBColor(128, 128, 128)
    p.alignment = PP_ALIGN.CENTER
    
    # Save the presentation
    prs.save('presentasi_office_archive.pptx')
    print("Presentasi berhasil dibuat: presentasi_office_archive.pptx")

if __name__ == "__main__":
    create_presentation()
